doc/design/context.md

```diff
@@ -0,0 +1,166 @@
+## Context Design
+
+
+A Net is executed by single or several threads. A Context is related to a thread and records necessary runtime resources.
+
+Context is defined as follows:
+
+```
+class Context {};
+```
+
+### CUDAContext
+
+Because the Tensor computation are executed by Eigen library, which needs an Eigen::GpuDevice type object as  parameter. And the GpuDevice parameter is constructed with an Eigen::CudaStreamDevice object. We need to set a specific GpuID and CudaStream to create a Eigen::CudaStream object.
```

Comment:
-> which needs an Eigen::GpuDevice type object as  **a** parameter

. And -> , and
. We -> , we

We need to set both a specific GpuID and an CudaStream to create a Eigen::CudaStream object.